gitlab/__init__.py

```diff
@@ -362,7 +362,7 @@ def _build_url(self, path):
         else:
             return '%s%s' % (self._url, path)
 
-    def http_request(self, verb, path, query_data={}, post_data={},
+    def http_request(self, verb, path, query_data={}, post_data=None,
```

Comment:
The default value of `post_data` for `http_post` and `http_put` are also an empty dict, I guess you should change the value to `None` for those methods as well (`http_request` is never called directly).

Reply:
I didn't read the latest code carefully. The version I used is 1.3. But what was causing problem is GET request with a body (by default will be b'{}'), which get rejected by caching servers. PUT and POST requests with a non empty body should be alright. There might be better ways or better places to make the change.

Reply:
Ah yes, sorry. So this looks good. Thanks!